integration-libs/cdc/core/store/effects/cdc-user-addresses.effect.spec.ts

```diff
@@ -110,40 +115,218 @@ describe('CDC User Addresses effect', () => {
     spyOn(userAddressService, 'getAddresses').and.returnValue(
       of(mockUserAddresses)
     );
+    spyOn(globalMessageService, 'remove');
+    spyOn(globalMessageService, 'add');
+    TestBed.compileComponents();
   });
 
   describe('cdcAddUserAddress$', () => {
-    it('should send default address to CDC on add addresses success', () => {
-      const action = new UserActions.AddUserAddressSuccess(mockUserAddresses);
+    it('should not update default address in CDC and show error message if add address fails', (done) => {
+      const action = hot('-a', {
+        a: new UserActions.AddUserAddressSuccess(mockUserAddress),
+      });
+      const error = {
+        status: 'ERROR',
+        errorMessage: 'Error adding default address in CDC',
+      };
+      const expected = cold('-');
+
+      cdcJSService.updateAddressWithoutScreenSet = createSpy().and.returnValue(
+        of(error)
+      );
+
+      actions$ = action;
+
+      cdcUserAddressesEffect.cdcAddUserAddress$.subscribe(() => {
+        expect(cdcJSService.updateAddressWithoutScreenSet).toHaveBeenCalled();
+        cdcJSService
+          .updateAddressWithoutScreenSet(mockUserAddress.formattedAddress + '')
+          .subscribe({
+            error: (error) => {
+              expect(globalMessageService.add).toHaveBeenCalledWith(
+                { key: error.errorMessage },
+                GlobalMessageType.MSG_TYPE_ERROR
+              );
+              done();
+            },
+          });
+        done();
+      });
```

Comment:
1. Do we _really_ need to subscribe to `cdcJSService.updateAddressWithoutScreenSet` in this test? Can't we simply put `expect(globalMessageService.add).toHaveBeenCalledWith(...)` in line 142, instead? IMHO  invoking `cdcJSService.updateAddressWithoutScreenSet()` manually and testing its output should not be a subject of an unit test of the ngrx effect. 
2. I believe in an unit should call `done()` callback only once - it's a signal to Jasmine that we're not waiting for anything more. Having 2 `done()` invocations in one test is a code smell - one of those invocations might be ignored and test will still pass. It makes sense to leave only one `done()` invocation - in a place the will be executed as the last one - in this case in the most nested subscription callback. (But please see point 1. above - I'm really hoping we don't need this nested manual invocation of service's method and its manual subscription).

Similar remarks apply also to other tests in this file.

Note: remark (1) can be fixed after 6.0 FF. Remark (2) should be simple to fix - just leave only the nested `done`

Reply:
I created a sub-PR to this branch with refactoring marble tests for `cdcAddUserAddress$` effect only.
https://github.com/SAP/spartacus/pull/17049

@SherwinVarghese  Feel free to merge it and refactor yourself other marble tests (for update/delete effects) analogically.

You can do it in another PR, after 6.0 FF as well.